tests/config/test_workers.py

```diff
@@ -138,7 +139,9 @@ def test_transitional_configs_master(self) -> None:
         """
 
         main_process_config = self._make_worker_config(
-            worker_app="synapse.app.homeserver", worker_name=None
+            worker_app="synapse.app.homeserver",
+            worker_name=None,
+            extras={"instance_map": {"main": {"host": "127.0.0.1", "port": 0}}},
```

Comment:
Why did the tests need fixing up, if the original change was intended to be backwards compatible?

Reply:
The tests relied on the fact that they never used any functionality that talked to the master; the `worker_*` settings weren't set in the tests either. I find this a bug TBH since all non-master workers need to know where the master is (AFAIK?) so checking it at config time is actually a minor improvement.